tests/base/test_regularization.py

```diff
@@ -785,9 +786,14 @@ def mesh(self, request):
     def test_weights(self, mesh):
         """Test cell_weights and weights."""
         weights = np.ones(len(mesh))
+        weights_dict = {"weights": weights}
         msg = "Cannot simultaneously pass 'weights' and 'cell_weights'."
         with pytest.raises(ValueError, match=msg):
-            BaseRegularization(mesh, weights=weights, cell_weights=weights)
+            BaseRegularization(
+                mesh,
+                weights=weights_dict["weights"],
+                cell_weights=weights_dict["weights"],
```

Comment:
Just passing the dictionaries should be enough (no need top ass the array inside it).
```suggestion
                weights=weights_dict,
                cell_weights=weights_dict,
```